src/index.js

```diff
@@ -11,19 +11,17 @@ export function h(name, attributes /*, ...rest*/) {
       for (length = node.length; length--; ) {
         rest.push(node[length])
       }
-    } else if (node != null && node !== true && node !== false) {
```

Comment:
@frenzzy Why would you remove this line here?

Reply:
Component execution was moved away from `h` function but they also may return `null` and boolean values. We have to check the result of component call, so I added this check to `app` (`getVNode)`. And just to check the same thing twice I removed this line.

Reply:
@frenzzy But calling a component function will call `h`, so you could leave this here too.

Reply:
```js
// no h() call here
const Component = (attributes, children) => null
const Component = ({ condition }) => condition && <div />
const Component = ({ condition }) => condition ? <div /> : null
```
ref: [React: Preventing Component from Rendering](https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering)

Reply:
@frenzzy 

```js
h(Component)
```

Reply:
`h(Component)` in this PR returns `{ nodeName: Component, attributes: {}, children: [] }`. Inside patch function I execute `Component` and replace VNode with the result.
```js
const Component = () => result

<div><Component /></div>
// =>
h('div', {}, [h(Component)])
// =>
{ nodeName: 'div', attributes: {}, children: [
  { nodeName: Component, attributes: {}, children: [] }
] }
// =>
{ nodeName: 'div', attributes: {}, children: [
  result
] }
```